pkg/dns/service.go

```diff
@@ -0,0 +1,272 @@
+	}
+
+	if len(dnsRecords) != 0 {
+		for _, r := range dnsRecords {
+			managedHosts = append(managedHosts, r.Name)
+		}
+		return managedHosts, dnsRecords, AlreadyAssignedErr
+	}
+	log.Log.Info("no managed host found generating one")
+	hostKey := shortuuid.NewWithNamespace(t.GetNamespace() + t.GetName())
+	zones := getManagedZones()
+	var chosenZone zone
+	var managedHost string
+	for _, z := range zones {
+		if z.Default {
+			managedHost = strings.ToLower(fmt.Sprintf("%s.%s", hostKey, z.RootDomain))
+			chosenZone = z
+			break
+		}
+	}
+	if chosenZone.ID == "" {
+		return managedHosts, dnsRecords, fmt.Errorf("no zone available to use")
+	}
+	record, err := s.RegisterHost(ctx, managedHost, hostKey, chosenZone.DNSZone)
+	if err != nil {
+		log.Log.Error(err, "failed to register host ")
+		return managedHosts, dnsRecords, err
+	}
+	managedHosts = append(managedHosts, managedHost)
+	dnsRecords = append(dnsRecords, record)
+	return managedHosts, dnsRecords, nil
+}
+
+func (s *Service) RegisterHost(ctx context.Context, h string, id string, zone v1.DNSZone) (*v1.DNSRecord, error) {
+	dnsRecord := v1.DNSRecord{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      h,
+			Namespace: s.defaultCtrlNS,
+			Labels:    map[string]string{labelRecordID: id},
+		},
+	}
+
+	err := s.controlClient.Create(ctx, &dnsRecord, &client.CreateOptions{})
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
+		return nil, err
+	}
+	//host may already be present
+	if err != nil && k8serrors.IsAlreadyExists(err) {
+		err = s.controlClient.Get(ctx, client.ObjectKeyFromObject(&dnsRecord), &dnsRecord)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return &dnsRecord, nil
+}
+
+// this is temporary and will be replaced in the future by CRD resources
+type zone struct {
+	v1.DNSZone
+	RootDomain string
+	Default    bool
+}
+
+// this is temporary and will be replaced in the future by CRD resources
+func getManagedZones() []zone {
+	return []zone{{
+		DNSZone:    v1.DNSZone{ID: os.Getenv("AWS_DNS_PUBLIC_ZONE_ID")},
+		RootDomain: os.Getenv("ZONE_ROOT_DOMAIN"),
+		Default:    true,
+	}}
+}
+
+// awsEndpointWeight returns the weight Value for a single AWS record in a set of records where the traffic is split
+// evenly between a number of clusters/ingresses, each splitting traffic evenly to a number of IPs (numIPs)
+//
+// Divides the number of IPs by a known weight allowance for a cluster/ingress, note that this means:
+// * Will always return 1 after a certain number of ips is reached, 60 in the current case (maxWeight / 2)
+// * Will return values that don't add up to the total maxWeight when the number of ingresses is not divisible by numIPs
+//
+// The aws weight value must be an integer between 0 and 255.
+// https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/resource-record-sets-values-weighted.html#rrsets-values-weighted-weight
+func awsEndpointWeight(numIPs int) string {
+	maxWeight := 120
+	if numIPs > maxWeight {
+		numIPs = maxWeight
```

Comment:
why not just return 1 here?

Reply:
again this was just copied over from GLBC